pkg/auth/dependency/webapp/authenticate_provider_impl.go

```diff
@@ -376,8 +362,22 @@ func (p *AuthenticateProviderImpl) HandleSSOCallback(w http.ResponseWriter, r *h
 		return
 	}
 	webappSSOState := SSOState(state.Extra)
-	callbackURL = webappSSOState.CallbackURL()
-	sid = webappSSOState.WebAppStateID()
+	requestQuery := webappSSOState.RequestQuery()
+	v, err := url.ParseQuery(requestQuery)
```

Comment:
`v` should be used in the `err != nill` part of `writeResponse`. So in case of error, `ui_locales` is preserved and the user can perform a retry.

Reply:
Updated @louischan-oursky